ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/PullQueryExecutorMetrics.java

```diff
@@ -0,0 +1,215 @@
+
+  private final List<Sensor> sensors;
+  private final Sensor localRequestsSensor;
+  private final Sensor remoteRequestsSensor;
+  private final Sensor latencySensor;
+  private final Sensor requestRateSensor;
+  private final Sensor errorRateSensor;
+  private final Metrics metrics;
+
+  public PullQueryExecutorMetrics() {
+    this.metrics = MetricCollectors.getMetrics();
+    this.sensors = new ArrayList<>();
+    this.localRequestsSensor = configureLocalRequestsSensor();
+    this.remoteRequestsSensor = configureRemoteRequestsSensor();
+    this.latencySensor = configureRequestSensor();
+    this.requestRateSensor = configureRateSensor();
+    this.errorRateSensor = configureErrorRateSensor();
+  }
+
+  @Override
+  public void close() {
+    sensors.forEach(sensor -> metrics.removeSensor(sensor.name()));
+  }
+
+  public void recordLocalRequests(final double value) {
+    this.localRequestsSensor.record(value);
+  }
+
+  public void recordRemoteRequests(final double value) {
+    this.remoteRequestsSensor.record(value);
+  }
+
+  public void recordRate(final double value) {
+    this.requestRateSensor.record(value);
+  }
+
+  public void recordLatency(final double value) {
+    this.latencySensor.record(value);
+  }
+
+  public void recordErrorRate(final double value) {
+    this.errorRateSensor.record(value);
+  }
+
+  private Sensor configureLocalRequestsSensor() {
+    final Sensor sensor = metrics.sensor(
+        PULL_QUERY_METRIC_GROUP + "-" + PULL_REQUESTS + "-local");
+    sensor.add(
+        metrics.metricName(
+            PULL_REQUESTS + "-local-count",
+            PULL_QUERY_METRIC_GROUP,
+            "Count of local pull query requests"
+        ),
+        new WindowedCount()
+    );
+    sensors.add(sensor);
+    return sensor;
+  }
+
+  private Sensor configureRemoteRequestsSensor() {
+    final Sensor sensor = metrics.sensor(
+        PULL_QUERY_METRIC_GROUP + "-" + PULL_REQUESTS + "-remote");
+    sensor.add(
+        metrics.metricName(
+            PULL_REQUESTS + "-remote-count",
+            PULL_QUERY_METRIC_GROUP,
+            "Count of remote pull query requests"
+        ),
+        new WindowedCount()
+    );
+    sensors.add(sensor);
+    return sensor;
+  }
+
+  private Sensor configureRateSensor() {
+    final Sensor sensor = metrics.sensor(
+        PULL_QUERY_METRIC_GROUP + "-" + PULL_REQUESTS + "-rate");
+    sensor.add(
+        metrics.metricName(
+            PULL_REQUESTS + "-rate",
+            PULL_QUERY_METRIC_GROUP,
+            "Rate of pull query requests"
+        ),
+        new Rate()
+    );
+    sensors.add(sensor);
+    return sensor;
+  }
+
+  private Sensor configureErrorRateSensor() {
+    final Sensor sensor = metrics.sensor(
+        PULL_QUERY_METRIC_GROUP + "-" + PULL_REQUESTS + "-error-rate");
+    sensor.add(
+        metrics.metricName(
+            PULL_REQUESTS + "-error-rate",
+            PULL_QUERY_METRIC_GROUP,
+            "Rate of erroneous pull query requests"
```

Comment:
It'd be great if we can also add the user-specified custom metric tags (https://github.com/confluentinc/ksql/blob/7bf5c120c7128d4825f74fc7e691ef2c68c6911b/ksqldb-common/src/main/java/io/confluent/ksql/util/KsqlConfig.java#L128) to these metrics, similar to how we do for engine metrics (https://github.com/confluentinc/ksql/blob/7bf5c120c7128d4825f74fc7e691ef2c68c6911b/ksqldb-engine/src/main/java/io/confluent/ksql/internal/KsqlEngineMetrics.java#L309)